src/components/listing-search-page/SearchResults/index.jsx

```diff
@@ -22,54 +31,88 @@ import {
 function ListingSearch() {
   const [keyWord, setKeyWord] = useState(window.location.href.split('=')[1]);
   const [cardData, setCardData] = useState(null);
+  const [resultCount, setResultCount] = useState(0);
+  const inputValue = useRef();
   const navigate = useNavigate();
 
+  const sorryMessage = `We did not find listings for: ${keyWord}, Edit or remove these filters for
+  best results.`;
+
   const fetchListingDataBySearchKey = async (address) => {
     const listingData = await getListingsBySearchKey(address);
-    setCardData(listingData);
 
-    const appPaths = {
-      listingPaths: {
-        search: '/listing/search',
-      },
-    };
+    setCardData(listingData);
+    setResultCount(`${listingData.length} Results Found in`);
 
-    navigate(`${appPaths.listingPaths.search}?key=${address}`);
+    navigate(`${listingPaths.search}?key=${address}`);
   };
 
   useEffect(() => {
     if (keyWord) fetchListingDataBySearchKey(keyWord);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const handleInputClose = () => {
+    setKeyWord('');
+    inputValue.current.value = '';
+    fetchListingDataBySearchKey(inputValue.current.value);
+    inputValue.current.focus();
+  };
+
   return (
     <ListingSearchContainer>
       <SearchContainer>
-        <input
-          type="text"
-          defaultValue={keyWord}
-          onChange={(e) => {
-            setKeyWord(e.target.value);
-          }}
-        />
+        <InputWrapper>
+          <input
+            ref={inputValue}
+            type="text"
+            defaultValue={keyWord}
+            onChange={(e) => {
+              setKeyWord(e.target.value);
+            }}
+            onKeyDown={(e) => {
+              if (e.keyCode === 13) {
+                setKeyWord(e.target.value);
+                fetchListingDataBySearchKey(e.target.value);
+              }
+            }}
+          />
+          {keyWord ? <ClearIcon onClick={handleInputClose} /> : null}
+        </InputWrapper>
         <FilterButton>
           <FilterIcon />
           <p>Filter</p>
         </FilterButton>
-        <Button
+        <SearchButton
           onClick={() => {
             fetchListingDataBySearchKey(keyWord);
           }}
         >
           <SearchIcon />
-        </Button>
+        </SearchButton>
       </SearchContainer>
 
-      <CardsContainer>
-        {cardData?.map((item) => (
-          <Card data={item} key={item.id} />
-        ))}
-      </CardsContainer>
+      {cardData?.length ? (
+        <>
+          <ResultCount>
+            <ResultNumber>{resultCount}</ResultNumber>
+            <ResultText>{keyWord}</ResultText>
+          </ResultCount>
```

Comment:
This component change in satet
```suggestion
          <ResultCountContainer>
            <ResultNumber>{results.?length ? results.?length: 0 }</ResultNumber>
            <ResultText>{keyWord}</ResultText>
          </ResultCountContainer>
```
Can you please explain to me why are you using state for component management here?